### ISSUE TITLE
Error on `:Gcommit`

### ISSUE THREAD

**Issue by teoljungberg** [tpope/vim-fugitive#1103]
I've ran into this a couple of times for the last few weeks, but I have not found the culprit. The only commonality I've found is when I use `gvim(1)` and have different tabs for different directories.

Sometimes, when I run `:Gcommit` I get the following error thrown:

```
:Gcommit
Error detected while processing function <SNR>67_Commit:
line   30:
E676: No matching autocommands for acwrite buffer  
```

Which points me to https://github.com/tpope/vim-fugitive/blob/9fec8aadc54372cbea6ac3c9cf6909054140b573/autoload/fugitive.vim#L1998

Any ideas as to what the issue could be?

**Comment by tpope** [comment]
Fugitive doesn't have any `acwrite` buffers. Where is that coming from?

**Comment by tpope** [comment]
Yeah this has to be `'autowrite'` paired with a broken buffer from some other plugin.

**Comment by teoljungberg** [comment]
I narrowed it down. It has to do with `:terminal` buffers.

If you open a file and modify it, then open a `:terminal` window - leave it running and switch back to the original file and perform `:Gcommit %` - the error is thrown.

**Comment by tpope** [comment]
Ah. The error is also thrown if you `:!ls`. Perhaps it's something we can work around but I think the first step would be to investigate it as a Vim bug.

**Comment by teoljungberg** [comment]
It looks like it's related to https://github.com/vim/vim/issues/3391. And it appears to be patched in `8.1.334`. I've only experienced this my MacVim instance, and it's running `8.1.280`.
